phoebe:    
    joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 100


    dual_arm_controller:
      type: "position_controllers/JointTrajectoryController"
      joints:
        - left_arm_shoulder_roll_joint
        - left_arm_shoulder_pitch_joint
        - left_arm_shoulder_yaw_joint
        - left_arm_elbow_pitch_joint
        - left_arm_elbow_yaw_joint
        - left_arm_wrist_pitch_joint
        - left_arm_wrist_roll_joint
        - right_arm_shoulder_roll_joint
        - right_arm_shoulder_pitch_joint
        - right_arm_shoulder_yaw_joint
        - right_arm_elbow_pitch_joint
        - right_arm_elbow_yaw_joint
        - right_arm_wrist_pitch_joint
        - right_arm_wrist_roll_joint
      
      constraints:
        goal_time: &goal_time_constraint 5.0
        stopped_velocity_tolerance: &stopped_velocity_constraint 20.0 # FIXME: See #8429
        left_arm_shoulder_roll_joint:
          goal: &goal_pos_constraint 0.2
        left_arm_shoulder_pitch_joint:
          goal: *goal_pos_constraint
        left_arm_shoulder_yaw_joint:
          goal: *goal_pos_constraint    
        left_arm_elbow_pitch_joint:
          goal: *goal_pos_constraint
        left_arm_elbow_yaw_joint:
          goal: *goal_pos_constraint
        left_arm_wrist_pitch_joint:
          goal: *goal_pos_constraint
        left_arm_wrist_roll_joint:
          goal: *goal_pos_constraint 
          
        right_arm_shoulder_roll_joint:
          goal: *goal_pos_constraint
        right_arm_shoulder_pitch_joint:
          goal: *goal_pos_constraint
        right_arm_shoulder_yaw_joint:
          goal: *goal_pos_constraint
        right_arm_elbow_pitch_joint:
          goal: *goal_pos_constraint
        right_arm_elbow_yaw_joint:
          goal: *goal_pos_constraint
        right_arm_wrist_pitch_joint:
          goal: *goal_pos_constraint
        right_arm_wrist_roll_joint:
          goal: *goal_pos_constraint  
      stop_trajectory_duration: 0.0 
     

    # left_arm_controller:
      # type: "position_controllers/JointTrajectoryController"
      # joints:
        # - left_arm_shoulder_roll_joint
        # - left_arm_shoulder_pitch_joint
        # - left_arm_shoulder_yaw_joint
        # - left_arm_elbow_pitch_joint
        # - left_arm_elbow_yaw_joint
        # - left_arm_wrist_pitch_joint
        # - left_arm_wrist_roll_joint



    # right_arm_controller:
      # type: "position_controllers/JointTrajectoryController"
      # joints:    
        # - right_arm_shoulder_roll_joint
        # - right_arm_shoulder_pitch_joint
        # - right_arm_shoulder_yaw_joint
        # - right_arm_elbow_pitch_joint
        # - right_arm_elbow_yaw_joint
        # - right_arm_wrist_pitch_joint
        # - right_arm_wrist_roll_joint


    left_gripper_controller:
      #type: "position_controllers/JointTrajectoryController"
      type:  position_controllers/GripperActionController
      #type: "position_controllers/JointPositionController"
      joint: left_arm_gripper_joint
      
      constraints:
        goal_time: *goal_time_constraint
        stopped_velocity_tolerance: *stopped_velocity_constraint
        left_arm_gripper_joint:
          goal: *goal_pos_constraint
        

      stop_trajectory_duration: 0.0



    right_gripper_controller:
      #type: "position_controllers/JointTrajectoryController"
      type:  position_controllers/GripperActionController
      #type: "position_controllers/JointPositionController"
      joint: right_arm_gripper_joint
      
      constraints:
        goal_time: *goal_time_constraint
        stopped_velocity_tolerance: *stopped_velocity_constraint
        right_arm_gripper_joint:
          goal: *goal_pos_constraint
        

      stop_trajectory_duration: 0.0
       


    wheels_controller:
        type: "diff_drive_controller/DiffDriveController"
        publish_rate: 50
        left_wheel: wheel_left_joint
        right_wheel: wheel_right_joint
        wheel_separation: 0.31
        
        wheel_radius : 0.095
        
    # Odometry covariances for the encoder output of the robot. These values should
    # be tuned to your robot's sample odometry data, but these values are a good place
    # to start
     #   pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
     #   twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
        
        
        pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
        twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

    # Top level frame (link) of the robot description
        base_frame_id: base_footprint

    # Velocity and acceleration limits for the robot
        # linear:
           # x:
            # has_velocity_limits    : true
            # max_velocity           : 3   # m/s
            # has_acceleration_limits: true
            # max_acceleration       : 1   # m/s^2
        # angular:
           # z:
            # has_velocity_limits    : true
            # max_velocity           : 2.0   # rad/s
            # has_acceleration_limits: true
            # max_acceleration       : 6.0   # rad/s^2

        wheel_separation_multiplier: 1.0 # default: 1.0
        wheel_radius_multiplier    : 1.0 # default: 1.0

      # Velocity commands timeout [s], default 0.5
        cmd_vel_timeout: 0.25

      # Velocity and acceleration limits
      # Whenever a min_* is unspecified, default to -max_*
        linear:
         x:
          has_velocity_limits    : true
          max_velocity           :  1.0 # m/s
          min_velocity           : -0.5 # m/s
          has_acceleration_limits: true
          max_acceleration       :  0.8 # m/s^2
          min_acceleration       : -0.4 # m/s^2
        angular:
         z:
          has_velocity_limits    : true
          max_velocity           : 1.7 # rad/s
          has_acceleration_limits: true
          max_acceleration       : 1.5 # rad/s^2        

    ptu_controller:
      type: "position_controllers/JointTrajectoryController"
      joints:
        - ptu_pan_joint
        - ptu_tilt_joint
        
      constraints:
        
        ptu_pan_joint:
          goal: *goal_pos_constraint 
        ptu_tilt_joint:
          goal: *goal_pos_constraint

      stop_trajectory_duration: 0.0      